data/src/main/java/com/linkedin/data/codec/ProtobufDataCodec.java

```diff
@@ -365,15 +377,16 @@ public void doubleValue(double value) throws IOException
     public void stringValue(String value) throws IOException
     {
       int symbolId;
-      if (_symbolTable != null && (symbolId = _symbolTable.getSymbolId(value)) != SymbolTable.UNKNOWN_SYMBOL_ID)
+      if ((symbolId = _symbolTable.getSymbolId(value)) != SymbolTable.UNKNOWN_SYMBOL_ID)
       {
         _protoWriter.writeByte(STRING_REFERENCE_ORDINAL);
         _protoWriter.writeUInt32(symbolId);
       }
       else
       {
-        _protoWriter.writeByte(STRING_LITERAL_ORDINAL);
-        _protoWriter.writeString(value);
+        // If the byte length is the same as the string length, then this is an ASCII-only string.
+        _protoWriter.writeString(value, byteLength ->
```

Comment:
Then I have to make it a bifunction since I cannot use the string inline. Prefer to keep it this way for that reason. The comment anyways clearly indicates what the function is doing.